src/login/pam_systemd.c

```diff
@@ -48,6 +48,13 @@
 #include "terminal-util.h"
 #include "util.h"
 #include "path-util.h"
+#include "cgroup-util.h"
+
+/* PAM data resource names */
+const char *LIMIT_MEMORY_MAX = "systemd.memory_max";
+const char *LIMIT_TASKS_MAX  = "systemd.tasks_max";
+const char *LIMIT_CPU_WEIGHT = "systemd.cpu_weight";
+const char *LIMIT_IO_WEIGHT  = "systemd.io_weight";
```

Comment:
the above allocate both the strings as immutable data in the resulting ELF object, as well as a mutable pointer to each. That's not really desirable. Why not just use `#define LIMIT_MEMORY MAX "systemd.memory_max"` instead? that way, you only add the string itself to the immutable data section of the ELF object.

That said, why have this a all? introducing symbolic names for things makes sense if what they resolve to is hard to handle or remember otherwise or is subject to change. But neither is the case here, you just map one name to pretty much a slightly differently written version of itself. Hence I figure, just using the literal string everywhere makes things simpler to read and preferable, hence I'd do away with any symbolic definitions like the above. In particular as you use each definition exactly once...